examples_tests/0.ImportanceSamplingEnvMaps/main.cpp

```diff
@@ -0,0 +1,668 @@
+			luminanceImageRegions->begin()->imageOffset = { 0, 0, 0 };
+			luminanceImageRegions->begin()->imageExtent = { luminanceImageParams.extent.width, luminanceImageParams.extent.height, 1 };
+
+			core::smart_refctd_ptr<ICPUImage> luminanceImage = ICPUImage::create(std::move(luminanceImageParams));
+			luminanceImage->setBufferAndRegions(core::smart_refctd_ptr(luminancePdfBuffer), luminanceImageRegions);
+
+			// Create out image
+			const size_t conditionalCdfBufferSize = pdfDomainExtent.X * pdfDomainExtent.Y * sizeof(double);
+			core::smart_refctd_ptr<ICPUBuffer> conditionalCdfBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(conditionalCdfBufferSize);
+			memset(conditionalCdfBuffer->getPointer(), 0, conditionalCdfBufferSize);
+
+			auto conditionalCdfImageParams = luminanceImage->getCreationParameters();
+			conditionalCdfImageParams.format = EF_R64_SFLOAT;
+
+			auto conditionalCdfImageRegions(luminanceImageRegions);
+			conditionalCdfImage = ICPUImage::create(std::move(conditionalCdfImageParams));
+			conditionalCdfImage->setBufferAndRegions(std::move(conditionalCdfBuffer), conditionalCdfImageRegions);
+
+			// Set up the filter state
+			SATFilter sum_filter;
+			SATFilter::state_type state;
+
+			state.inImage = luminanceImage.get();
+			state.outImage = conditionalCdfImage.get();
+			state.inOffset = { 0, 0, 0 };
+			state.inBaseLayer = 0;
+			state.outOffset = { 0, 0, 0 };
+			state.outBaseLayer = 0;
+			state.extent = luminanceImage->getCreationParameters().extent;
+			state.layerCount = luminanceImage->getCreationParameters().arrayLayers;
+			state.scratchMemoryByteSize = state.getRequiredScratchByteSize(state.inImage, state.extent);
+			state.scratchMemory = reinterpret_cast<uint8_t*>(_NBL_ALIGNED_MALLOC(state.scratchMemoryByteSize, 32));
+			state.axesToSum = ((0) << 2) | ((0) << 1) | ((1) << 0); // ZYX
+			state.inMipLevel = 0;
+			state.outMipLevel = 0;
+
+			if (!sum_filter.execute(std::execution::par_unseq, &state))
+				std::cout << "SAT filter failed for some reason" << std::endl;
+
+			_NBL_ALIGNED_FREE(state.scratchMemory);
+
+			// From the outImage you gotta extract integrals and normalize
+			double* conditionalCdfPixel = (double*)conditionalCdfImage->getBuffer()->getPointer();
+
+			conditionalIntegrals = core::make_smart_refctd_ptr<ICPUBuffer>(pdfDomainExtent.Y * sizeof(double));
+			double* conditionalIntegralsPixel = (double*)conditionalIntegrals->getPointer();
+			for (uint32_t y = 0; y < pdfDomainExtent.Y; ++y)
+				*conditionalIntegralsPixel++ = conditionalCdfPixel[y * pdfDomainExtent.X + (pdfDomainExtent.X - 1)];
+
+			conditionalCdfPixel = (double*)conditionalCdfImage->getBuffer()->getPointer();
+			conditionalIntegralsPixel = (double*)conditionalIntegrals->getPointer();
+
+			// now normalize
+			for (uint32_t y = 0; y < pdfDomainExtent.Y; ++y)
+			{
+				for (uint32_t x = 0; x < pdfDomainExtent.X; ++x)
+				{
+					conditionalCdfPixel[y * pdfDomainExtent.X + x] /= conditionalIntegralsPixel[y];
+				}
+			}
```

Comment:
could be folded into SAT filter (it already has a `Normalize` boolean)
`/= SAT_VALUES[SCRATCH_ADDRESS(sumX ? MaxX:x,sumY ? y:maxY,sumZ ? z:maxZ)]`

would also "fix" the filter normalization option after your axis-summing choice